### ISSUE TITLE
Double parking "P"

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#2230]
**[Submitted to the original trac issue database at 4.53pm, Thursday, 27th August 2009]**

Look here:

http://www.openstreetmap.org/?lat=45.356127&lon=9.65366&zoom=18&layers=B000FTF

there is a parking with 2 "P". I try to regenerate the tiles locally with R17299 and I don't get the double "P"

**Comment by tomhughes** [comment]
**Author: tom@compton.nu**
**[Added to the original trac issue at 5.32pm, Thursday, 27th August 2009]**

Jon - do we still have the old hack to add P markers installed? If we do I guess it is clashing with the new code to render markers for areas...

**Comment by tomhughes** [comment]
**Author: Ldp**
**[Added to the original trac issue at 11.46am, Friday, 28th August 2009]**

Yes, that one is still active (or actually, in osm2pgsql's source). I was actually counting on it picking about the same spot as the polygon centroid mapnik is supposed to take, dropping one of them due to a collision.

I wonder which P comes from which source. If the out of center one is mapnik's, why would it choose such an awkward spot for such a simple polygon?

**Comment by tomhughes** [comment]
**Author: jburgess777@googlemail.com**
**[Added to the original trac issue at 12.03pm, Friday, 28th August 2009]**

The two nodes on the right have been moved several times. If they were aligned with the road to right then the second P would be in about the centre of this area. I don't believe the diff update process does anything to delete the generated P nodes when parking areas are updated.

There are currently two generated P nodes in the DB:
```
gis=> select osm_id,amenity,astext(transform(way,4326)) from planet_osm_point where osm_id=26267629;
  osm_id  | amenity |                  astext
----------+---------+------------------------------------------
 26267629 | parking | POINT(9.65349861966763 45.3561133580288)
 26267629 | parking | POINT(9.65370460336228 45.3561374715661)
(2 rows)
```
I have now deleted them both. This only effects the Mapnik rendering DB, not the OSM ways or nodes.
